### ISSUE TITLE
Feature Request: Allow passing the credentials object itself instead of service names

### ISSUE THREAD

**Issue by aktivalux** [SAP/leonardo-iot-sdk-nodejs#4]
The SDK needs the credentials to authenticate the requests. Currently those credentials can only be indirectly provided by passing a service name (regular service binding or user provided service) to the client constructor. This is great if you are in a Cloud Foundry / XSA environment. But as an SDK this library should be also agnostic to the environment.

Hence it would be great to provide a way to pass your own credentials object instead of just a service name. This way the SDK can be also easily used outside of the SAP cloud platform.

I could imagine that the constructor can take either a string (status quo) or an object in the form of:

```js
{
   // either provide credentials
   credentials: {
      host: 'foo.bar',
      ...
   },
   // or a service name - shorthand would be to just pass as string with
   // the service name below instead of this object
   serviceName: 'my-foo-service'
}
```

This approach could also maintain backward compatability as well as provide means for future extensions (e.g. register pre/post request hooks, ...) by adding additional properties for the object.

**Comment by Magoli1** [comment]
Hey Sven,
thank you for opening an issue. :blush:
There is actually a way to provide the credentials via a `default-env.json` file (more info [here](https://github.com/SAP/leonardo-iot-sdk-nodejs#2-setup-authorization-for-local-usage)). Is this sufficient for your needs? If not, could you please elaborate a little more about your use-case? 

BR Robert

**Comment by aktivalux** [comment]
Think outside SAP and outside of cloud foundry (cf)/XSA. Maybe one wants to use your SDK in an application running on K8s, or in a AWS Lambda function. In those environments you might have  your own way to configure your applications. E.g. you could be using popular libraries such as "[config](https://github.com/lorenwest/node-config)" or "[rc](https://github.com/dominictarr/rc)" for Node.js.

Now when using those libraries there is no straight way to provide the configuration to the SDK. You have to take the detour and write the config to the environment variables (VCAP_SERVICES) before initializing this SDK (or to disk in the form of a default-env.json file). You are kind of forcing the user to implement the configuration the "Cloud-Foundry-Way" even though he might not be using cloud foundry.

tl;dr: For an SDK I'd consider it a good thing if it makes as few assumptions on your runtime environment as possible. Providing convenience functions (such as a shorthand in which you just provide the cf service name) for certain environments however is totally fine.

**Comment by Magoli1** [comment]
Hey Sven,

thanks for the additional clarification! We are internally evaluating the idea at the moment. Feel free to open a PR :smiley:

BR Robert

**Comment by soerenbuehler** [comment]
Hi Sven,

we added the support of manual credential configuration within the client instantiation with Version 0.1.3 (already in master), please checkout the sample here: https://github.com/SAP/leonardo-iot-sdk-nodejs#authorization-concept

Please feel free to use any of the suggested frameworks to load the specific configuration into your runtime for separation between dev, test and productive configuration.

Kind regards
Sören
